apis/v1alpha2/gateway_types.go

```diff
@@ -0,0 +1,835 @@
+// valid. Some invalid configurations can be caught synchronously via a
+// webhook, but there are many cases that will require asynchronous
+// signaling via the GatewayStatus block.
+type GatewaySpec struct {
+	// GatewayClassName used for this Gateway. This is the name of a
+	// GatewayClass resource.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	GatewayClassName string `json:"gatewayClassName"`
+
+	// Listeners associated with this Gateway. Listeners define
+	// logical endpoints that are bound on this Gateway's addresses.
+	// At least one Listener MUST be specified.
+	//
+	// An implementation MAY group Listeners by Port and then collapse each
+	// group of Listeners into a single Listener if the implementation
+	// determines that the Listeners in the group are "compatible". An
+	// implementation MAY also group together and collapse compatible
+	// Listeners belonging to different Gateways.
+	//
+	// For example, an implementation might consider Listeners to be
+	// compatible with each other if all of the following conditions are
+	// met:
+	//
+	// 1. Either each Listener within the group specifies the "HTTP"
+	//    Protocol or each Listener within the group specifies either
+	//    the "HTTPS" or "TLS" Protocol.
+	//
+	// 2. Each Listener within the group specifies a Hostname that is unique
+	//    within the group.
+	//
+	// 3. As a special case, one Listener within a group may omit Hostname,
+	//    in which case this Listener matches when no other Listener
+	//    matches.
+	//
+	// If the implementation does collapse compatible Listeners, the
+	// hostname provided in the incoming client request MUST be
+	// matched to a Listener to find the correct set of Routes.
+	// The incoming hostname MUST be matched using the Hostname
+	// field for each Listener in order of most to least specific.
+	// That is, exact matches must be processed before wildcard
+	// matches.
+	//
+	// If this field specifies multiple Listeners that have the same
+	// Port value but are not compatible, the implementation must raise
+	// a "Conflicted" condition in the Listener status.
+	//
+	// Support: Core
+	//
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=64
+	Listeners []Listener `json:"listeners"`
+
+	// Addresses requested for this gateway. This is optional and
+	// behavior can depend on the GatewayClass. If a value is set
+	// in the spec and the requested address is invalid, the
+	// GatewayClass MUST indicate this in the associated entry in
+	// GatewayStatus.Addresses.
+	//
+	// If no Addresses are specified, the GatewayClass may
+	// schedule the Gateway in an implementation-defined manner,
+	// assigning an appropriate set of Addresses.
+	//
+	// The GatewayClass MUST bind all Listeners to every
+	// GatewayAddress that it assigns to the Gateway.
+	//
+	// Support: Core
```

Comment:
What does `Support: Core` mean?

Reply:
Covered in more detail here: https://gateway-api.sigs.k8s.io/concepts/guidelines/#conformance